[Logs] Let the agent run logs-agent

tmichelet opened the pull request:
### What does this PR do?

- Add logs-agent binary on linux
- Add a logs-agent check that starts the logs agent

### Motivation

We'd like to include the logs-agent as part of agent 6 to dogfood it and for beta customers.

### Additional Notes

#### Related PRs

agent 5 PR - https://github.com/DataDog/dd-agent/pull/3520

#### Notes

The logs-agent is available for linux only. I am not sure what's the status of windows support for datadog-agent, thus haven't tested this scenario.

#### Tests

- gitlab build: [119045](https://gitlab.ddbuild.io/datadog/datadog-agent/pipelines/119045) contains updated builds. I set the [artifact](https://gitlab.ddbuild.io/datadog/datadog-agent/-/jobs/1639040/artifacts/file/.omnibus/pkg/datadog-agent6_0.0.0.20170915130649-1_amd64.deb) to be kept for a longer period

##### Tests the logs-agent included but disabled on a vm

- provision a `ubuntu/trusty64` vagrant box
- download the deb package, found at 
- configure the datadog-agent: in `/etc/datadog-agent/datadog.yaml`
  ```
  dd_url: https://app.datadoghq.com
  api_key: <api_key>
  confd_path: "/etc/datadog-agent/conf.d"
  ```
- install the package: `sudo dpkg -i datadog-agent6_0.0.0.20170915130649-1_amd64.deb`
- assert it reports metrics in datadog
- assert logs-agent is not running: `grep logs-agent /var/log/datadog/agent.log`

##### Tests the logs-agent included and enabled on a vm

- start from previous step
- setup proper configuration for the logs agent. See [internal doc](https://docs.google.com/document/d/1CNQg2hNaIn0WNJT9kTXuQAnhUO7qaeepaxpLclzBtfg/edit#heading=h.do7z5blfmc73) or ping #ramen. You should setup `/etc/datadog-agent/conf.d/logs-agent.yaml` and `/etc/datadog-agent/conf.d/logs-integrations.yaml`
  `/etc/datadog-agent/conf.d/logs-agent.yaml`
  ```
  init_config:

  instances:
    - whatever: anything

  log_dd_url: "intake.logs.datadoghq.com"
  logset: playground2
  api_key: <api_key>
  hostname: "tristan-host"
  ```
  `/etc/datadog-agent/conf.d/logs-integration.yaml`
  ```
  init_config:

  instances:
    - whatever: anything

  logs:
    - type: file
      path: /home/vagrant/test.log
      appname: bench-logs-agent
      source: bench-logs-agent
      logset: playground2
    - type: tcp
      logset: playground2
      source: java
      port: 10514
  ```
- restart dd-agent: `sudo service datadog-agent6 restart`
- assert logs-agent is up and running: `grep 'Starting logs-agent' /var/log/datadog/agent.log`
- assert that it works as expected by submitting logs lines:
  ```
  echo "hello" >> test.log
  grep 'Connecting' /var/log/datadog/agent.log
  2017/09/15 08:07:30 Connecting to the backend: intake.sheepdog.datad0g.com:10514 - skip_ssl_validation: true
  ```
  assert the log line appears in your logset

tmichelet commented:
@masci what's the next step regarding this PR? Should I rebase on new omnibus config?

tmichelet committed:
[Logs] Add core check that starts logs-agent

tmichelet committed:
[Logs] Add omnibus config to include logs agent in agent build

masci commented on omnibus/config/software/datadog-logs-agent.rb:
```diff
@@ -0,0 +1,20 @@
+name "datadog-logs-agent"
+always_build true
+
+logs_agent_version = ENV['logs_agent_version']
```
Just an headsup, we're trying to remove any env var access from the software definitions to have more predictable builds (omnibus has its own way to inject a version from the outside).

masci reviewed (dismissed):
LGTM!
Left a comment on the omnibus software def but we can change it later

tmichelet committed:
[Logs] Build - always build alpha version of logs-agent

we're trying to remove any env var access from the software
definitions to have more predictable builds
we're effectively always building that version, so let's remove
the option

tmichelet commented:
@masci updated.
Also, I don't see any artifact in circle-ci, where are those uploaded? (And did I miss any documentation around that?)

masci commented:
The release pipeline for v6 is entirely hosted on gitlab, you can find the artifacts there. The pipeline is not quite finished yet, no docs atm 😞

masci reviewed (approved):
LGTM!

tmichelet committed:
Merge branch 'master' into tristan/logs-agent
